### PR TITLE
Weighted metrics without metrics

### LINKED ISSUE #454: `weighted_metrics` not showing up next to training progress bar
Tried porting this example (https://keras.io/examples/generative/gpt2_text_generation_with_kerasnlp/) to Keras Core. `accuracy` (passed as a `weighted_metric`) doesn't show up in the training bar (across backends).

Reproduced it here (with JAX): https://colab.research.google.com/drive/1XByfu9-vaFLnXdUjW6tLypMYy36C8LVx?usp=sharing
This is with legacy Keras: https://colab.research.google.com/drive/1fucRLYK_7ZDMRN5GURMgRIe80JknaFhj?usp=sharing

### PR DESCRIPTION
**Pull request by mihirparadkar**
Fixes #454.

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM. Thank you!

**Comment by fchollet** [comment]
Please run `sh shell/format.sh`

**Review by fchollet — APPROVED** [review]
